engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/EngineJobExecutionActor.scala

```diff
@@ -153,11 +168,21 @@ class EngineJobExecutionActor(jobKey: BackendJobDescriptorKey,
     context.actorOf(EngineJobHashingActor.props(jobDescriptor, fileHasherActor, activity))
   }
 
-  def lookupCachedResult(jobDescriptor: BackendJobDescriptor, cacheResultId: MetaInfoId) = {
+  def lookupCachedResult(jobDescriptor: BackendJobDescriptor, taskOutputs: Seq[TaskOutput], cacheResultId: MetaInfoId) = {
     // TODO: Start up a backend job copying actor (if possible, otherwise just runJob). That should send back the BackendJobExecutionResponse
-    self ! FailedNonRetryableResponse(jobKey, new Exception("Call cache result copying not implemented!"), None)
+    //self ! FailedNonRetryableResponse(jobKey, new Exception("Call cache result copying not implemented!"), None)
+    val cachingSimpletonActor = context.actorOf(CachingSimpletonActor.props(cacheResultId, taskOutputs))
     // While the cache result is looked up, we wait for the response just like we were waiting for a Job to complete:
-    goto(RunningJob) using EJEAPartialCompletionData(None, None)
+    goto(PreparingCachedOutputs) using EJEAJobDescriptorData(Option(jobDescriptor), _)
+  }
+
+  def cacheJob(jobDescriptor: BackendJobDescriptor, cachedJobOutputs: JobOutputs, bjeaProps: Props) = {
+    val bjcaProps = bjeaProps
```

Comment:
We'd probably want the JobPreparation actor to return bjcaProps (in addition to the bjeaProps). I think it'd be awkward to crowbar cache copying into the same actor as job execution